### PR TITLE
docs: refresh CI docs

### LINKED ISSUE #12855: [BUG] Caching browser section looks obsolete and wrong
https://playwright.dev/docs/ci needs to be rewritten?

### PR DESCRIPTION
**Pull request by aslushnikov**
- Travis CI docs are badly rotten. I recall there was a need to
  list all the custom apt packages in the `travis.yml`, and it's a
  hassle to maintain this list for us. I remove the whole section for now
  since we don't see lots of Travis CI usage among our clients.
- Refresh other bits

Fixes #12855

### CODE REVIEW

File: docs/src/ci.md [hunk 1]
```diff
@@ -13,20 +13,20 @@ configurations for common CI providers.
...
    npm ci
    # or
    npm install
 
-   # Install Playwright browsers
-   npx playwright install
+   # Install Playwright browsers and dependencies
+   npx playwright install --with-deps
    ```
    ```bash python
```
Comment (dgozman) [inline docs/src/ci.md:27]: Let's add other languages!

File: docs/src/ci.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ steps:
     if: always()
     uses: actions/upload-artifact@v2
     with:
-      name: playwright-results
+      name: test-results
```
Comment (dgozman) [inline docs/src/ci.md:62]: playwright-report

File: docs/src/ci.md [hunk 3]
```diff
@@ -123,16 +121,18 @@ Suggested configuration
...
 
 ### Azure Pipelines
 
 For Windows or macOS agents, no additional configuration required, just install Playwright and run your tests.
 
-For Linux agents, you can use [our Docker container](./docker.md) with Azure Pipelines support for [running containerized jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=azure-devops). Alternatively, you can refer to the [Dockerfile](./docker.md) to see additional dependencies that need to be installed on a Ubuntu agent.
+For Linux agents, you can use [our Docker container](./docker.md) with Azure
+Pipelines support for [running containerized
+jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=azure-devops).
+Alternatively, you can use [Command line tools](./cli.md#install-system-dependencies) to install all necessary dependencies for Ubuntu 18.04
```
Comment (dgozman) [inline docs/src/ci.md:134]: Let's avoid `Ubuntu 18.04` because it will become stale. Just say "necessary dependencies"?

**Review by dgozman — APPROVED** [review]

File: docs/src/ci.md [hunk 4]
```diff
@@ -123,16 +121,18 @@ Suggested configuration
...
+1. Using `--init` Docker flag or [dumb-init](https://github.com/Yelp/dumb-init) is recommended to avoid special
+   treatment for processes with PID=1. This is a common reason for zombie processes.
 
 ### Azure Pipelines
 
 For Windows or macOS agents, no additional configuration required, just install Playwright and run your tests.
 
-For Linux agents, you can use [our Docker container](./docker.md) with Azure Pipelines support for [running containerized jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=azure-devops). Alternatively, you can refer to the [Dockerfile](./docker.md) to see additional dependencies that need to be installed on a Ubuntu agent.
+For Linux agents, you can use [our Docker container](./docker.md) with Azure
+Pipelines support for [running containerized
```
Comment (yury-s) [inline docs/src/ci.md:132]: drop "for" ?

File: docs/src/ci.md (hunk 3 above)
Comment (yury-s) [inline docs/src/ci.md:134]: for Ubuntu LTS (22.04 is coming soon)?

**Review by yury-s — APPROVED** [review]

**Pull request merged**